src/main/MainScreenWithTabs.js

```diff
@@ -10,6 +10,7 @@ export default class MainScreenWithTabs extends PureComponent<{}> {
   static contextTypes = {
     styles: () => null,
   };
+  static router = MainTabs.router; //Try to comment out this line and check navigation state Tab navigator routes are registered with this
```

Comment:
Here we give a router to MainScreenWithTabs(the one which MainTabs.router is owning) by doing this whenever parent navigator(AppNavigator) looks in MainScreenWithTabs it finds a router corresponding to the screen. Thus it gets to know there are children routes to MainScreenWithTabs